configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -628,7 +632,15 @@ public void createLocalSessionFromDistributedApplicationPackage(long sessionId)
             ApplicationId applicationId = sessionZKClient.readApplicationId()
                     .orElseThrow(() -> new RuntimeException("Could not find application id for session " + sessionId));
             log.log(Level.FINE, () -> "Creating local session for tenant '" + tenantName + "' with session id " + sessionId);
-            createLocalSession(sessionDir, applicationId, sessionId);
+            try {
+                createLocalSession(sessionDir, applicationId, sessionId);
+            } catch (RuntimeException e) {
```

Comment:
Isn't there a lock we can hold instead?

Reply:
Yes, that would be better, I'll look into it